src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+        {
+            Debug.Assert(manager != null);
+            Debug.Assert(targetMethod.IsDefinition);
+
+            return new MethodGroupConversionCacheFrame(manager, targetMethod, CalculateNeededArity(targetMethod));
+        }
+
+        private static int CalculateNeededArity(MethodSymbol targetMethod)
+        {
+            var arity = 1; // the type of the delegate
+
+            arity += targetMethod.Arity;
+
+            for (var containingType = targetMethod.ContainingType; containingType!= null; containingType = containingType.ContainingType)
+            {
+                arity += containingType.Arity;
+            }
+
+            return arity;
+        }
+
+        public static ImmutableArray<TypeSymbol> GetTypeArgumentsFromConversion(TypeSymbol conversionType, MethodSymbol targetMethod)
+        {
+            var typeArgumentsBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
+
+            typeArgumentsBuilder.Add(conversionType);
+
+            if (targetMethod.Arity > 0)
+            {
+                var args = ((ConstructedMethodSymbol)targetMethod).TypeArguments;
+                AddTypeArgumentsReversed(typeArgumentsBuilder, ref args);
+            }
+
+            for (var containingType = targetMethod.ContainingType; containingType != null; containingType = containingType.ContainingType)
+            {
+                if (containingType.Arity > 0)
+                {
+                    var args = containingType.TypeArguments;
+                    AddTypeArgumentsReversed(typeArgumentsBuilder, ref args);
+                }
+            }
+
+            typeArgumentsBuilder.ReverseContents();
+            return typeArgumentsBuilder.ToImmutableAndFree();
+        }
+
+        private static void AddTypeArgumentsReversed(ArrayBuilder<TypeSymbol> builder, ref ImmutableArray<TypeSymbol> args)
+        {
+            var i = args.Length;
+            while (i-- > 0)
+            {
+                builder.Add(args[i]);
+            }
+        }
+
+        public override ImmutableArray<Symbol> GetMembers()
+        {
+            if ( DelegateField != null )
+            {
+                return ImmutableArray.Create<Symbol>(DelegateField);
+            }
+
+            return ImmutableArray<Symbol>.Empty;
+        }
+
+        internal override void AddSynthesizedAttributes(ModuleCompilationState compilationState, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            base.AddSynthesizedAttributes(compilationState, ref attributes);
+
+            var compilation = ContainingSymbol.DeclaringCompilation;
+            Debug.Assert(compilation != null, "MethodGroupConversionCacheFrame is not contained in a source module?");
+
+            AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(
+                WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
```

Comment:
Copied from `SynthesizedContainer`, as per the discussion above, this code will be deleted when the frame is defined as synthesized container. But this attribute would still get generated.